components/apimgt/org.wso2.carbon.apimgt.common.gateway/src/main/java/org/wso2/carbon/apimgt/common/gateway/jwtgenerator/AbstractAPIMgtGatewayJWTGenerator.java

```diff
@@ -104,13 +105,19 @@ public String generateToken(JWTInfoDto jwtInfoDto) throws JWTGeneratorException
 
     public String buildHeader() throws JWTGeneratorException {
         String jwtHeader = null;
+        X509Certificate x509Certificate = (X509Certificate) jwtConfigurationDto.getPublicCert();
 
         if (NONE.equals(signatureAlgorithm)) {
             StringBuilder jwtHeaderBuilder = new StringBuilder();
             jwtHeaderBuilder.append("{\"typ\":\"JWT\",");
             jwtHeaderBuilder.append("\"alg\":\"");
             jwtHeaderBuilder.append(JWTUtil.getJWSCompliantAlgorithmCode(NONE));
             jwtHeaderBuilder.append('\"');
+            if (jwtConfigurationDto.useKid()) {
```

Comment:
Missed this section

Reply:
Fixed via https://github.com/wso2/carbon-apimgt/pull/12091/commits/cb7270216c230dd9aa6f54e8d7f592774fd113ef